File: src/main/java/com/header/header/domain/reservation/entity/Reservation.java
package com.header.header.domain.reservation.entity;

import com.header.header.domain.reservation.converter.ReservationStateConverter;
import com.header.header.domain.reservation.dto.BasicReservationDTO;
import com.header.header.domain.reservation.enums.ReservationState;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.sql.Date;
import java.sql.Time;

@Entity
@Table(name="tbl_reservation")
@Getter
@NoArgsConstructor( access = AccessLevel.PROTECTED)
//@DynamicInsert //값이 null인 필드 자동 제외하여 default값 반영
public class Reservation {

    @Id
    @GeneratedValue( strategy = GenerationType.IDENTITY)
    private Integer resvCode;
    private Integer userCode;
    private Integer shopCode;
    private Integer menuCode;
    private Date resvDate;
    private Time resvTime;
    private String userComment;

    @Convert(converter = ReservationStateConverter.class)
    private ReservationState resvState;

    public void cancelReservation() {

        this.resvState = ReservationState.CANCEL;
    }

    public void modifyReservation(BasicReservationDTO reservationDTO){
        this.menuCode = reservationDTO.getMenuCode();
        this.resvDate = reservationDTO.getResvDate();
        this.resvTime = reservationDTO.getResvTime();
        this.userComment = reservationDTO.getUserComment();
    }

}

File: llm/app/models/reservation.py
# reservation.py
from sqlalchemy import Table, Column, Integer, String, Date, Time, ForeignKey, MetaData
from app.core.db import metadata  # 이미 metadata를 import했다고 가정

Reservation = Table(
    'tbl_reservation',  # 실제 테이블 이름과 맞춰야 함
    metadata,
    Column('resv_code', Integer, primary_key=True, autoincrement=True),
    Column('user_code', Integer, ForeignKey('tbl_user.user_code'), nullable=False),
    Column('shop_code', Integer, ForeignKey('tbl_shop.shop_code'), nullable=False),
    Column('menu_code', Integer, ForeignKey('tbl_menu.menu_code'), nullable=False),
    Column('resv_date', Date, nullable=False),
    Column('resv_time', Time, nullable=False),
    Column('user_comment', String(255), nullable=True),
    Column('resv_state', String(20), nullable=False, default='예약확정'),
)

